geyser-plugin-manager/src/transaction_notifier.rs

```diff
@@ -90,12 +90,117 @@ impl TransactionNotifierImpl {
         transaction_status_meta: &'a TransactionStatusMeta,
         transaction: &'a SanitizedTransaction,
     ) -> ReplicaTransactionInfoV2<'a> {
+        let msg = transaction.message();
+        let instructions = msg.instructions();
+        let account_keys = msg.account_keys();
+
         ReplicaTransactionInfoV2 {
             index,
             signature,
-            is_vote: transaction.is_simple_vote_transaction(),
+            is_vote: if !account_keys.is_empty() && !instructions.is_empty() {
```

Comment:
We need to review further. As I see the following already did similar thing:

```
/// Simple vote transaction meets these conditions:
/// 1. has 1 or 2 signatures;
/// 2. is legacy message;
/// 3. has only one instruction;
/// 4. which must be Vote instruction;
#[inline]
pub fn is_simple_vote_transaction_impl<'a>(
    signatures: &[Signature],
    is_legacy_message: bool,
    mut instruction_programs: impl Iterator<Item = &'a Pubkey>,
) -> bool {
    signatures.len() < 3
        && is_legacy_message
        && instruction_programs
            .next()
            .xor(instruction_programs.next())
            .map(|program_id| program_id == &solana_sdk_ids::vote::ID)
            .unwrap_or(false)
}
```

See if we can utilize that.